hive-metastore/src/main/java/org/apache/iceberg/hive/MetastoreClientVersion.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import org.apache.hive.common.util.HiveVersionInfo;
+
+public enum MetastoreClientVersion {
+  HIVE_4(4),
+  HIVE_3(3),
+  HIVE_2(2),
+  HIVE_1_2(1),
+  NOT_SUPPORTED(0);
+
+  private final int order;
+  private static final MetastoreClientVersion current = calculate();
+
+  MetastoreClientVersion(int order) {
+    this.order = order;
+  }
+
+  public static MetastoreClientVersion current() {
+    return current;
+  }
+
+  public static boolean min(MetastoreClientVersion other) {
+    return current.order >= other.order;
+  }
+
+  private static MetastoreClientVersion calculate() {
+    String version = HiveVersionInfo.getShortVersion();
+    String[] versions = version.split("\\.");
+    switch (versions[0]) {
+      case "4":
+        return HIVE_4;
+      case "3":
+        return HIVE_3;
+      case "2":
+        return HIVE_2;
+      case "1":
+        if (versions[1].equals("2")) {
+          return HIVE_1_2;
+        } else {
+          return NOT_SUPPORTED;
```

Comment:
Why are we explicitly not-supporting versions lesser than Hive 1.2 now?  I don't think there was any checks   previously to throw an exception for this case.  Anyway, we are protecting the agentInfo code around Hive 2 check

Reply:
We support Hive 1.2 through Spark. I think we do not want to start supporting even older versions of Hive. That is the reason that I explicitly kept Hive 1.2 here and added anything else as unsupported.

Reply:
Makes sense, I am not sure if its desirable to make the effort here to explicitly throw an exception though for Hive less than 1.2.  Also was there an explicit note in any Iceberg docs?  

Though im open if thats what community feels, any thoughts: @rdblue @RussellSpitzer

Reply:
Is is never called out explicitly, but it would be hard find Hive older than 1.2 in any production environment. (I know of one, but that is a serious outlier)

Reply:
Yea makes sense, chatted offline, this is more for code completeness purpose as NOT_SUPPORTED is not used anywhere.  Somehow I mistakenly thought this was going throw an exception.  Im ok with this then.